modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -673,6 +689,84 @@ def _union_all(
 
         return new_frame
 
+    def _join_by_index(self, other_modin_frames, how, sort, ignore_index):
+        if how == "outer":
+            raise NotImplementedError("outer join is not supported in OmniSci engine")
+
+        lhs = self._maybe_materialize_rowid()
+        reset_index_names = False
+        for rhs in other_modin_frames:
+            rhs = rhs._maybe_materialize_rowid()
+            if len(lhs._index_cols) != len(rhs._index_cols):
```

Comment:
Why is join done by `_index_cols` but not by `_index_cache`?

Reply:
Join is done in OmniSci and `_index_cols` holds OmniSci table columns holding an index. `_index_cache` holds built `pandas.Index` only for executed frames.